Support AGT Counter driver for Renesas RA6, RA4, RA2 devices

khoa-nguyen-18 opened the pull request:
Add support for AGT Counter driver for RA6, RA4, RA2 devices

- Add AGT Counter driver support on RA6M1, RA6M2, RA6M3, RA6M4, RA6M5, RA6E1, RA4M2, RA4M3, RA4W1, RA2A1
- Add support `samples/drivers/counter/alarm` for RA6M1, RA6M2, RA6M3, RA6M4, RA6M5, RA6E1, RA4M2, RA4M3, RA4W1, RA2A1

soburi commented on boards/renesas/ek_ra6m1/doc/index.rst:
```diff
@@ -94,6 +94,8 @@ The below features are currently supported on Zephyr OS for EK-RA6M1 board:
 +-----------+------------+----------------------+
 | CLOCK     | on-chip    | clock control        |
 +-----------+------------+----------------------+
+| COUNTER   | on-chip    | counter              |
```
Wrong commit message, should start with `boards: renesas:...`

khoa-nguyen-18 replied:
Thank you, I have changed the commit message and split this modification into a commit

soburi commented on dts/arm/renesas/ra/ra6/r7fa6e2bx.dtsi:
```diff
@@ -7,6 +7,13 @@
 #include <arm/renesas/ra/ra6/ra6-cm33-common.dtsi>
 #include <zephyr/dt-bindings/clock/ra_clock.h>
 
+/delete-node/ &agt0;
```
It's better to split the changes in the dts folder into a separate commit.

khoa-nguyen-18 replied:
Thank you, I have split this dts modification into a commit. Could you please re-check again?

khoa-nguyen-18 committed:
dts: arm: renesas: Add AGT counter support for RA6, RA4, RA2

- Add dts node to support AGT counter for:
ra6-cm4, ra6-cm33 (eccept r7fa6e2bx),
ra4-cm4, ra4-cm33 (eccept r7fa4e2b93cfm),
ra2xx.

Signed-off-by: Khoa Nguyen <khoa.nguyen.xh@renesas.com>

khoa-nguyen-18 committed:
boards: renesas: Add counter doc for RA6, RA4, RA2

Add doc for AGT counter for:
ek_ra6m1, ek_ra6m2, ek_ra6m3, ek_ra6m4, ek_ra6m5, fpb_ra6e1,
ek_ra4w1, ek_ra4m2, ek_ra4m3, ek_ra2a1.

Signed-off-by: Khoa Nguyen <khoa.nguyen.xh@renesas.com>

khoa-nguyen-18 committed:
samples: drivers: counter: Support test alarm for RA6, RA4, RA2

Add overlay to support test alarm for:
ek_ra6m1, ek_ra6m2, ek_ra6m3, ek_ra6m4, ek_ra6m5, fpb_ra6e1,
ek_ra4w1, ek_ra4m2, ek_ra4m3, ek_ra2a1.

Signed-off-by: Khoa Nguyen <khoa.nguyen.xh@renesas.com>

soburi reviewed (dismissed):
LGTM

soburi commented:
@nordic-krch 
Could you take a look please?

khoa-nguyen-18 commented:
I just rebased main to solve the conflict. @soburi @KhiemNguyenT @nordic-krch Sorry for the inconvenience, can you please take a look?
